src/System Application/App/AppSource Gallery/src/AppSourceProductManager.Codeunit.al

```diff
@@ -0,0 +1,604 @@
+        Product.PublisherDisplayName := CopyStr(GetStringValue(Offer, 'publisherDisplayName'), 1, MaxStrLen(Product.PublisherDisplayName));
+        Product.PublisherType := CopyStr(GetStringValue(Offer, 'publisherType'), 1, MaxStrLen(Product.PublisherType));
+        Product.RatingAverage := GetDecimalValue(Offer, 'ratingAverage');
+        Product.RatingCount := GetIntegerValue(Offer, 'ratingCount');
+        Product.ProductType := CopyStr(GetStringValue(Offer, 'productType'), 1, MaxStrLen(Product.ProductType));
+        Product.Popularity := GetDecimalValue(Offer, 'popularity');
+        Product.LastModifiedDateTime := GetDateTimeValue(Offer, 'lastModifiedDateTime');
+
+        Product.AppID := ExtractAppIDFromUniqueProductID(Product.UniqueProductID);
+
+        // Insert, if it fails to insert due to the data (ex duplicate ids), ignore the error
+        if not Product.Insert() then;
+    end;
+
+    [NonDebuggable]
+    local procedure GetAPIKey(): SecretText
+    var
+        ApiKey: SecretText;
+    begin
+        Init();
+        if not Dependencies.EnvironmentInformation_IsSaas() then
+            Error(NotSupportedOnPremisesErrorLbl);
+
+        Dependencies.AzureKeyVault_GetAzureKeyVaultSecret('MS-AppSource-ApiKey', ApiKey);
+        exit(ApiKey);
+    end;
+
+    local procedure Init()
+    begin
+        if not IsDependenciesSet then
+            SetDefaultDependencyImplementation();
+    end;
+
+    local procedure SetDefaultDependencyImplementation()
+    var
+        dependencyInstance: Codeunit "AppSource Product Manager";
+    begin
+        dependencies := dependencyInstance;
+        IsDependenciesSet := true;
+    end;
+
+    var
+        Dependencies: Interface "IAppSource Product Manager Dependencies";
+        IsDependenciesSet: boolean;
+        CatalogProductsUriLbl: label 'https://catalogapi.azure.com/products', Locked = true;
+        CatalogApiVersionQueryParamNameLbl: label 'api-version', Locked = true;
+        CatalogApiVersionQueryParamValueLbl: label '2023-05-01-preview', Locked = true;
+        CatalogApiOrderByQueryParamNameLbl: label '$orderby', Locked = true;
+        CatalogMarketQueryParamNameLbl: label 'market', Locked = true;
+        CatalogLanguageQueryParamNameLbl: label 'language', Locked = true;
+        CatalogApiFilterQueryParamNameLbl: Label '$filter', Locked = true;
+        CatalogApiSelectQueryParamNameLbl: Label '$select', Locked = true;
+        AppSourceListingUriLbl: Label 'https://appsource.microsoft.com/%1/product/dynamics-365-business-central/%2', Comment = '%1=Language, %2=Url Query Content', Locked = true;
+        AppSourceUriLbl: Label 'https://appsource.microsoft.com/%1/marketplace/apps?product=dynamics-365-business-central', Comment = '%=Language', Locked = true;
+        UnsupportedMarketNotificationLbl: Label 'Market %1 is not supported by AppSource. Defaulting to en. Change region in the user profile to change the market.', Comment = '%1=Market';
```

Comment:
%1 = country code, ex. en, de, us or gb.
The translation team has no idea what Market means. Same in other places. Locked strings of course don't need comments as comments are mainly for the translation team.

Reply:
To address in this PR